packages/engine/src/worlds/operator/execute.rs

```diff
@@ -17,28 +17,44 @@ pub async fn execute(
     // Even though we have epochs forcing timeouts within WASI
     // we still need to set a timeout on the host side since we need to cancel sleeping components too
     // see https://github.com/bytecodealliance/wasmtime-go/issues/233#issuecomment-2356238658
-    tokio::time::timeout(Duration::from_secs(deps.time_limit_seconds), {
-        let service_id = service_id.clone();
-        let workflow_id = workflow_id.clone();
-        async move {
-            crate::bindings::operator::world::WavsWorld::instantiate_async(
-                deps.store.as_operator_mut(),
-                &deps.component,
-                deps.linker.as_operator_ref(),
-            )
-            .await
-            .map_err(EngineError::Instantiate)?
-            .call_run(deps.store.as_operator_mut(), &input)
-            .await
-            .map_err(|e| match e.downcast_ref::<Trap>() {
-                Some(t) if *t == Trap::OutOfFuel => EngineError::OutOfFuel(service_id, workflow_id),
-                Some(t) if *t == Trap::Interrupt => EngineError::OutOfTime(service_id, workflow_id),
-                _ => EngineError::ComponentError(e),
-            })?
-            .map_err(EngineError::ExecResult)
-            .map(|r| r.map(|r| r.into()))
+    let responses: Vec<WasmResponse> =
+        tokio::time::timeout(Duration::from_secs(deps.time_limit_seconds), {
+            let service_id = service_id.clone();
+            let workflow_id = workflow_id.clone();
+            async move {
+                crate::bindings::operator::world::WavsWorld::instantiate_async(
+                    deps.store.as_operator_mut(),
+                    &deps.component,
+                    deps.linker.as_operator_ref(),
+                )
+                .await
+                .map_err(EngineError::Instantiate)?
+                .call_run(deps.store.as_operator_mut(), &input)
+                .await
+                .map_err(|e| match e.downcast_ref::<Trap>() {
+                    Some(t) if *t == Trap::OutOfFuel => {
+                        EngineError::OutOfFuel(service_id, workflow_id)
+                    }
+                    Some(t) if *t == Trap::Interrupt => {
+                        EngineError::OutOfTime(service_id, workflow_id)
+                    }
+                    _ => EngineError::ComponentError(e),
+                })?
+                .map_err(EngineError::ExecResult)
+                .map(|r| r.into_iter().map(|r| r.into()).collect())
+            }
+        })
+        .await
+        .map_err(|_| EngineError::OutOfTime(service_id, workflow_id))??;
+
+    // Invariant: If there are multiple responses, they must all have an event id salt
+    if responses.len() > 1 {
+        for response in &responses {
+            if response.event_id_salt.is_none() {
+                return Err(EngineError::MissingEventIdSalt);
+            }
         }
```

Comment:
I think we should also validate for duplicate `event_id_salt`, not only a missing one.